### PR TITLE
KNIDEPLOY-4221: Support running over IPV6

### PR DESCRIPTION
**Pull request by pawanpinjarkar**
Ref: https://issues.redhat.com/browse/KNIDEPLOY-4221
This PR is to Inject CA into the assisted-service pod. The changes in this PR are based on the knowledge documented in the below 2 docs -

1. https://docs.google.com/document/d/1oG87nmTr-pAz_yr81tXc8um4OLkPQDMOULC_zCxc5jM/edit#heading=h.cx2nv4cfuvqa
2. https://docs.google.com/document/d/1JN_KHsBpBk6vrf_aQjP9-vpmwODM5WcoJm18-k0Ofb4/edit

For deploying assisted service IPV6 cluster on top of the OCP cluster ( created through dev-scripts), we need to mirror the images to the local registry because there is no internet connectivity. We also need to create a new release by running `oc adm release new` and not just mirror it. More details :point_down: from link 2 above 

> During the first steps of the installation, the assisted-service will try to extract the openshift installer from the release, which is stored in the disconnected registry and server using a self-signed certificate, so we need a method for the assisted-service pods to trust this registry serving certificate.
> 
> In this example, a configmap is created inside the assisted-installer namespace containing a CA bundle that includes every common CA and the CA used to sign the certificate registry. Then, it’s mounted into the pod as a volume.

This PR is for creating a new config map and to volume mount it into assisted service pod.

The other steps mentioned in document link 1 above such as injecting ignition before downloading the discovery ISO, setting API and Ingress VIP, etc were performed manually through an API call.

### CODE REVIEW

**Comment by pawanpinjarkar** [comment]
/assign @jparrill 
This PR is for creating the config map required to deploy assisted service on IPV6 based on your doc -  [Inject CA into assisted-service pods](https://docs.google.com/document/d/1JN_KHsBpBk6vrf_aQjP9-vpmwODM5WcoJm18-k0Ofb4/edit#heading=h.sacp69wt8jj4). Can you please suggest if it is good to go or should I hold onto it? If there is any other approach to be followed, please let me know.

File: Makefile [hunk 1]
```diff
@@ -8,6 +8,7 @@ BUILD_TYPE := $(or ${BUILD_TYPE},standalone)
 TARGET := $(or ${TARGET},minikube)
 PROFILE := $(or $(PROFILE),minikube)
 KUBECTL=kubectl -n $(NAMESPACE)
+INSTALL_TYPE := $(or $(INSTALL_TYPE),IPV4)
```
Comment (filanov) [inline Makefile:11]: maybe network type?

File: Makefile [hunk 2]
```diff
@@ -219,6 +220,9 @@ deploy-service-requirements: | deploy-namespace deploy-inventory-service-file
 		--ocp-versions '$(subst ",\",$(OPENSHIFT_VERSIONS))' --public-registries "$(PUBLIC_CONTAINER_REGISTRIES)" \
 		--check-cvo $(CHECK_CLUSTER_VERSION) --apply-manifest $(APPLY_MANIFEST) $(ENABLE_KUBE_API_CMD) $(E2E_TESTS_CONFIG) \
 		--ipv6-support $(IPV6_SUPPORT) --enable-sno-dnsmasq $(ENABLE_SINGLE_NODE_DNSMASQ)
+	@if [ $(INSTALL_TYPE) = "IPV6" ]; then\
```
Comment (filanov) [inline Makefile:223]: maybe instead using the env here you can pass it as a parameter to the script?

**Review by filanov — COMMENTED** [review]
Just making sure that i understand, this CA is used by the service, how is it related to https://issues.redhat.com/browse/MGMT-4223 ? 
And how will it be used by the operator?

**Comment by pawanpinjarkar** [comment]
@filanov I will make the necessary changes for the operator. I have some questions for which I will reach out to @jparrill

**Comment by yevgeny-shnaidman** [comment]
@pawanpinjarkar where do we inject the mirror registries configuration into assisted-service? the release image can be modified to point directly to the mirror registry, but other images used by asissted-service will still point to the quay.io/ocpmetal

**Comment by yevgeny-shnaidman** [comment]
@pawanpinjarkar since the decision was made that we will use one mirror registry configuration, and it will be provided via assisted-service deployment, you will also need to add registries.conf contents during deployment
@mhrivnak what is the best format for it? mapping a congfigmap into a volume that will contain the whole registries.conf contents?

**Comment by flaper87** [comment]
It looks like @yevgeny-shnaidman is now working on this here https://github.com/openshift/assisted-service/pull/1563 

Is this correct? Should this PR be closed in favor of https://github.com/openshift/assisted-service/pull/1563 ?

**Comment by yevgeny-shnaidman** [comment]
> It looks like @yevgeny-shnaidman is now working on this here #1563
> 
> Is this correct? Should this PR be closed in favor of #1563 ?

yes, i took over the task, @pawanpinjarkar can you close your PR?

**Pull request closed without merging**